sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java

```diff
@@ -433,6 +433,7 @@ void beginTxn() {
         ProducerSpEL.beginTransaction(producer);
       }
 
+      @SuppressWarnings("FutureReturnValueIgnored")
```

Comment:
In general I think leaving `TODOs` in code is a poor practice. 
How about I open a jira task to review future handling in `KafkaIO`, informing that we've suppressed warnings (doesn't change current behaviour) but would like to review if we may drop messages on exceptional cases? I'll be happy to dig deeper with you on that.

Reply:
Makes sense. Please file a jira. I wanted to add the comment soon after this gets merged. I know why it is ok to ignore here (it will be waited on inside `commitTxn()`).  You can add the explanation in PR itself. If you are adding the comment, I suggest changing sendRecord() return the future and let the call ignore it with a comment saying they are flushed in commitTxn().

Reply:
I'll make the code change - new commit later